contracts/RelayHub.sol

```diff
@@ -152,16 +154,27 @@ contract RelayHub is RelayHubApi {
 	// for contract-specific checks.
 	// returns "0" if the relay is valid. other values represent errors.
 	// values 1..10 are reserved for can_relay. other values can be used by accept_relayed_call of target contracts.
-    function can_relay(address relay, address from, RelayRecipient to, bytes memory transaction, uint transaction_fee, uint gas_price, uint gas_limit, uint nonce, bytes memory approval) public view returns(uint32) {
-        bytes memory packed = abi.encodePacked("rlx:", from, to, transaction, transaction_fee, gas_price, gas_limit, nonce, address(this));
+    function can_relay(address relay, address from, RelayRecipient to, bytes memory encoded_function, uint transaction_fee, uint gas_price, uint gas_limit, uint nonce, bytes memory approval) public view returns(uint32) {
+        bytes memory packed = abi.encodePacked("rlx:", from, to, encoded_function, transaction_fee, gas_price, gas_limit, nonce, address(this));
         bytes32 hashed_message = keccak256(abi.encodePacked(packed, relay));
         bytes32 signed_message = keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", hashed_message));
         if (!GsnUtils.checkSig(from, signed_message,  approval))  // Verify the sender's signature on the transaction
             return 1;   // @from hasn't signed the transaction properly
         if (nonces[from] != nonce)
             return 2;   // Not a current transaction.  May be a replay attempt.
         // XXX check @to's balance, roughly estimate if it has enough balance to pay the transaction fee.  It's the relay's responsibility to verify, but check here too.
-        return to.accept_relayed_call(relay, from, transaction, gas_price, transaction_fee, approval); // Check to.accept_relayed_call, see if it agrees to accept the charges.
+        bytes memory transaction = abi.encodeWithSelector(to.accept_relayed_call.selector, relay, from, encoded_function, gas_price, transaction_fee, approval);
```

Comment:
'transaction' is the most generic word in the Smart Contract code ever. Call it 't'. Or, 'a'. Or, don't.

Reply:
Changed it to 'x'. 
We good?